cmd/secrets-provider/main.go

```diff
@@ -51,63 +51,52 @@ func main() {
 		printErrorAndExit(messages.CSPFK009E)
 	}
 
-	// Configure exponential backoff
-	expBackoff := backoff.NewExponentialBackOff()
-	expBackoff.InitialInterval = 2 * time.Second
-	expBackoff.RandomizationFactor = 0.5
-	expBackoff.Multiplier = 2
-	expBackoff.MaxInterval = 15 * time.Second
-	expBackoff.MaxElapsedTime = 2 * time.Minute
-
+	// Configure constant backoff for application container
+	constantBackOff := backoff.NewConstantBackOff(time.Duration(secretsConfig.SyncIntervalTime) * time.Minute)
+	count := 0
 	err = backoff.Retry(func() error {
```

Comment:
i may be missing something but it seems you are not doing anything with the returned `err` object. what happens if an error is returned somewhere inside the backoff loop? will this block still run:
```
/ Deleting the retrieved Conjur access token in case we got an error after retrieval.
			// if the access token is already deleted the action should not fail
			err = accessToken.Delete()
			if err != nil {
				log.Error(messages.CSPFK003E, err)
			}
			log.Error(messages.CSPFK038E)
			printErrorAndExit(messages.CSPFK039E)
```

Reply:
I just removed the `err =` 
if there is an error in the backoff loop, it will try again and again until that if is true
```
if count > secretsConfig.RetryCount {
}
```
Once the above is true we will process will exit with 1 error code and the AccessToken will be deleted and "Secrets Provider for Kubernetes failed to update Kubernetes Secrets" will be added to logs

Some background, I needed a way to break out of the loop because and that above if statement does just that